packages/template-retail-react-app/app/components/recommended-products/index.jsx

```diff
@@ -197,8 +197,8 @@ const RecommendedProducts = ({zone, recommender, products, title, shouldFetch, .
                 isFavourite: wishlist?.customerProductListItems?.some(
                     (item) => item.productId === product?.productId
                 ),
-                onFavouriteToggle: (isFavourite) => {
-                    const action = isFavourite ? removeItemFromWishlist : addItemToWishlist
+                onFavouriteToggle: (toBeFavourite) => {
+                    const action = toBeFavourite ? addItemToWishlist : removeItemFromWishlist
```

Comment:
By renaming the parameter, I think this should be clearer that it's the _target_ favourite state (not the current state). So we don't get confused whether to add or remove the item.